langchain-core/src/utils/testing/index.ts

```diff
@@ -212,30 +214,96 @@ export class FakeChatModel extends BaseChatModel {
   }
 }
 
-export class FakeStreamingChatModel extends BaseChatModel {
-  sleep?: number = 50;
+export class FakeStreamingChatModel extends BaseChatModel<FakeStreamingChatModelCallOptions> {
+  sleep: number = 50;
 
-  responses?: BaseMessage[];
+  responses: BaseMessage[] = [];
+
+  chunks: AIMessageChunk[] = [];
+
+  toolStyle: "openai" | "anthropic" | "bedrock" | "google" = "openai";
 
   thrownErrorString?: string;
 
-  constructor(
-    fields: {
-      sleep?: number;
-      responses?: BaseMessage[];
-      thrownErrorString?: string;
-    } & BaseLLMParams
-  ) {
-    super(fields);
-    this.sleep = fields.sleep ?? this.sleep;
-    this.responses = fields.responses;
-    this.thrownErrorString = fields.thrownErrorString;
+  private tools: StructuredTool[] = [];
+
+  constructor({
+    sleep = 50,
+    responses = [],
+    chunks = [],
+    toolStyle = "openai",
+    thrownErrorString,
+    ...rest
+  }: FakeStreamingChatModelFields & BaseLLMParams) {
+    super(rest);
+    this.sleep = sleep;
+    this.responses = responses;
+    this.chunks = chunks;
+    this.toolStyle = toolStyle;
+    this.thrownErrorString = thrownErrorString;
   }
 
   _llmType() {
     return "fake";
   }
 
+  bindTools(tools: StructuredTool[]) {
+    const merged = [...this.tools, ...tools];
+
+    const toolDicts = merged.map((t) => {
+      switch (this.toolStyle) {
+        case "openai":
+          return {
+            type: "function",
+            function: {
+              name: t.name,
+              description: t.description,
+              parameters: zodToJsonSchema(t.schema),
```

Comment:
tool schema are not always zod - tools can be defined as JSON schema

Reply:
Hi @benjamincburns , thanks for pointing out this. I'm now convertin only Zod schemas with `toJsonSchema` helper (existing) and plain JSON schemas are forwarded untouched.